File: Chapter 04/Python code/4_12.py

n=int(input("Enter 1st number: "))
m=int(input("Enter 2nd number: "))
lcm=1
for i in range(min(n,m),1, -1):
    if n>=i and m >=i:
        if n%i ==0 and m%i == 0:
            n=n/i
            m=m/i
            lcm=lcm*i
    else:
        break
if n!=0:
    lcm = lcm*n
if m!=0:
    lcm = lcm*m
print(lcm)

File: Chapter 04/C Code/4_12.c

#include<stdio.h>
#include<math.h>
#define min(n,m) n<m?n:m

void main()
{
int n,m,lcm,i;
printf("Enter 2 numbers: ");
scanf("%d%d", &n, &m);
lcm=1;
for (i=min(n,m); i>1; i--)
{
if (n>=i  && m >=i)
{
if (n%i ==0 && m%i == 0)
{n=n/i;
m=m/i;
lcm=lcm*i;
}
}
else
break;
}
if (n!=0)
lcm = lcm*n;
if (m!=0)
lcm = lcm*m;
printf("LCM is %d", lcm);
}
